pkg/networkservice/common/heal/client.go

```diff
@@ -121,12 +121,21 @@ func (f *healClient) eventloop(recv networkservice.MonitorConnection_MonitorConn
 
 			case networkservice.ConnectionEventType_UPDATE:
 				for _, conn := range event.GetConnections() {
-					f.connections[conn.GetId()] = conn
+					path := conn.GetPath()
+					index := int(path.GetIndex() - 1)
+					if index >= 0 && len(path.GetPathSegments()) > index {
+						// TODO - do we really need the conn here, and even if we do, is this the *right* conn?
+						f.connections[path.GetPathSegments()[index].Id] = conn.Clone()
+					}
```

Comment:
Heal needs to correctly translate the changes reported by downstream to upstream